File: examples/chapter10/shootbad.m
%%
% We compute shooting solutions for several values of $\lambda$.
lambda = (6:4:18)';
for k = 1:4
    lam = lambda(k);
    lval = -1;  rval = 0;
    phi = @(x,u,dudx) lam^2*u + lam^2;
    [x,u] = shoot(phi,[0,1],lval,[],rval,[],0);

    plot(x,u,'-'), hold on
    
    xlabel('x'), ylabel('u(x)'),  ylim([-1 0.5])   % ignore this line
    title('Shooting instability')    % ignore this line
end
legend(num2str(lambda),'location','northwest')   % ignore this line


%%
% The solutions do not stay negative on the right half of the domain as
% $\lambda$ increases.

File: julia/scripts/chapter10/shootbad.jl

include("../FNC.jl")
using Plots

lambda = (6:4:18)
for lam in lambda
    lval = -1.0;  rval = 0.0;
    phi = (x,u,dudx) -> lam^2*u + lam^2
    x,u = FNC.shoot(phi,(0.0,1.0),lval,[],rval,[],0.0)
    plot!(x,u,m=:o,label="\$\\lambda=$lam\$")
end
plot!([],[],label="",
    xaxis=("x"),yaxis=([-2,0.5],"u(x)"),title="Shooting instability",leg=:topleft)
